src/movies.js

```diff
@@ -1,14 +1,64 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array) {
+
+    const allDirectors = array.map(elm => elm.director)
+    
+    return allDirectors
+    
+}
+
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(array) {
+    
+    const spielberg = array.filter(elm => {
+        elm.director.includes(`Spielberg`) && elm.genre.includes('Drama')
+    })
+    return spielberg.length
+}
+
+
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+
+function ratesAverage(array) {
+    
+    const sumRates = array.reduce((acc, elm) => {
+        const sum = acc + elm.rate
+        const avg = sum / array.length
+        return avg
+    }, 0)
+    return sumRates
+}
```

Comment:
Este es el rate de cada elemento y lo estás dividiendo directamente por el total de elementos.

Mejor saca el sumatorio y luego divide.
```suggestion

function ratesAverage(array) {
    
    const sumRates = array.reduce((acc, elm) => {
        const sum = acc + elm.rate
        
        return sum
    }, 0)
    return sumRates/array.length
}
```
Además tendrás que pasarlo a dos decimales .toFixed(2) y eso devolverlo a tipo number (toFixed convierte a string).
`return parseFloat((sumRates/array.length).toFixed(2))`